src/codegen.cpp

```diff
@@ -709,10 +709,11 @@ typedef struct {
 
 typedef struct {
     size_t len;
+    // variable length
     struct {
         int64_t isref;
         Function *f;
-    } data[];
+    } data[1];
```

Comment:
OK so I find [this](http://stackoverflow.com/questions/3711233/is-the-struct-hack-technically-undefined-behavior/) stackoverflow thread. And it seems that there basically isn't a ISO C++ conforming way to do this.

* The flexible member is invalid C++ syntax and causes warning under `-pedantic`.

* The single element array could technically be UB but `It's not clear if it's legal or portable, but it is rather popular.` and `does seem to work under all known implementations`.... (and also issue no warnings.)